conn.go

```diff
@@ -82,6 +83,11 @@ type ConnErrorHandler interface {
 	HandleError(conn *Conn, err error, closed bool)
 }
 
+// How many timeouts we will allow to occur before the connection is closed
+// and restarted. This is to prevent a single query timeout from killing a connection
+// which may be serving more queries just fine.
+const timeoutLimit = 10
```

Comment:
Globally is fine, as long as it is settable - somebody is bound to come along and want to tweak this.